packages/11ty/content/_data/eleventyComputed.js

```diff
@@ -119,5 +109,36 @@ module.exports = {
       nextPage: pages[currentPageIndex + 1],
       previousPage: pages[currentPageIndex - 1]
     }
+  },
+  /**
+   * Contributors with a `pages` property containing data about the pages they contributed to
+   */
+  publicationContributors: ({ config, publication, pages }) => {
+    const { contributor, contributor_as_it_appears } = publication
+    return contributor_as_it_appears 
+      ? contributor_as_it_appears 
+      : contributor
+      /**
+       * Filtering because there are duplicate contributors here
+       * in eleventyComputed but not elsewhere. WHY?
+       */
+      .filter((itemA, index, items) => items.findIndex((itemB) => itemB.id===itemA.id)===index)
+      .map((item) => {
+        const { pic } = item
+        item.imagePath = pic
+          ? path.join(config.params.imageDir, pic)
+          : null
+        item.pages = pages && pages.filter(
+          ({ data }) => {
+            if (!data.contributor) return
+            return Array.isArray(data.contributor)
+              ? data.contributor.find(
+                (pageContributor) => pageContributor.id === item.id
+              )
+              : data.contributor.id === item.id
+          }
+        )
+        return item
```

Comment:
@hbalenda refactoring to rewrite the nested `filter` `map` `filter` and ternaries as separate functions may improved readability though if we could understand and fix the duplicate contributors that too would help.

Reply:
yeah, worth figuring this out before merging. i will investigate